packages/aws-cdk/lib/api/aws-auth/credential-plugins.ts

```diff
@@ -23,9 +23,11 @@ import { PluginHost } from '../plugin/plugin';
 export class CredentialPlugins {
   private readonly cache: { [key: string]: PluginCredentialsFetchResult | undefined } = {};
   private readonly host: PluginHost;
+  private readonly ioHelper: IoHelper;
 
-  constructor(host?: PluginHost) {
-    this.host = host ?? PluginHost.instance;
+  constructor(host: PluginHost, ioHelper: IoHelper) {
```

Comment:
just forcing the call site to pass the plugin host explicitly (which is still `PluginHost.instance`)

Reply:
We chatted about this and this class seems to be only used by a very small number of mostly outdated projects.